rust-lang/cargo
# refactor: use custom error instead of anyhow

### What does this PR try to resolve?

This defines custom error types for `rustfix` crate, instead of `anyhow`.

### How should we test and review this PR?

Some error messages were slightly tweaked but shouldn't affect the behavior.

I am not particularly sure about the SemVer compatibility. Some functions' return type has changed from `Result<_, anyhow::Error>` to `Result<_, rustfix::Error>`. People already consume `anyhow::Error` should be able to use `rustfix::Error` without updating their code I think.
